src/main/java/sirius/biz/protocol/DelegateJournalData.java

```diff
@@ -0,0 +1,148 @@
+     * @param entityRef the entity reference under which the journal message will be created
+     */
+    public DelegateJournalData(BaseEntity<?> owner, @Nonnull BaseEntityRef<?, ?> entityRef) {
+        this(owner, entityRef::getIdAsString, () -> Mixing.getNameForType(entityRef.getType()));
+    }
+
+    /**
+     * Defines the supplier responsible to deliver the message to write in the protocol upon entity changes.
+     * <p>
+     * The supplier is called automatically as an {@link AfterSave} action. If the supplier is not set
+     * or delivers empty results, no protocol message is written.
+     * <p>
+     * The {@link EntityDescriptor#getLabel() label} of the entity will be automatically
+     * preppend to the generated message.
+     *
+     * @param changeMessageSupplier the supplier of the journal message
+     * @return the object itself for fluent calls.
+     */
+    public DelegateJournalData withChangeMessageSupplier(Supplier<String> changeMessageSupplier) {
+        this.changeMessageSupplier = changeMessageSupplier;
+        return this;
+    }
+
+    /**
+     * Defines the supplier responsible to deliver the message to write in the protocol upon entity deletes.
+     * <p>
+     * The supplier is called automatically as an {@link AfterDelete} action. If the supplier is not set
+     * or delivers empty results, no protocol message is written.
+     * <p>
+     * The {@link EntityDescriptor#getLabel() label} of the entity will be automatically
+     * preppend to the generated message.
+     *
+     * @param deleteMessageSupplier the supplier of the journal message
+     * @return the object itself for fluent calls.
+     */
+    public DelegateJournalData withDeleteMessageSupplier(Supplier<String> deleteMessageSupplier) {
+        this.deleteMessageSupplier = deleteMessageSupplier;
+        return this;
+    }
+
+    /**
+     * Sets the skip flag.
+     * <p>
+     * Calling this with <tt>true</tt>, will skip all changes performed on the referenced entity instance.
+     *
+     * @param silent <tt>true</tt> to skip the recording of all changes on the referenced entity instance,
+     *               <tt>false</tt> to re-enable.
+     */
+    public void setSilent(boolean silent) {
+        this.silent = silent;
+    }
+
+    @AfterSave
+    protected void onSave() {
+        createJournalEntry(changeMessageSupplier);
+    }
+
+    @AfterDelete
+    protected void onDelete() {
+        createJournalEntry(deleteMessageSupplier);
+    }
+
+    private void createJournalEntry(Supplier<String> messageSupplier) {
+        if (silent
+            || !Sirius.isFrameworkEnabled(Protocols.FRAMEWORK_JOURNAL)
+            || messageSupplier == null
+            || owner.isNew()) {
+            return;
+        }
+
+        try {
+            String message = messageSupplier.get();
+            if (Strings.isFilled(message)) {
+                JournalData.addJournalEntry(targetType.get(), targetId.get(), owner.toString(), message);
```

Comment:
using owner.toString here is wrong as for consistency reasons we'd have to use target.toString() - I'd rather kill this field entirely and only store the owner.toString() during a delete of the main entity so that we can search for vanished entities. Everything else can be found via the targetType/targetId

Reply:
Therefore JournalData#onDelete must be updated accordingly